network/manager.go

```diff
@@ -67,18 +67,18 @@ type NetworkManager interface {
 	AddExternalInterface(ifName string, subnet string) error
 
 	CreateNetwork(nwInfo *NetworkInfo) error
-	DeleteNetwork(networkId string) error
-	GetNetworkInfo(networkId string) (NetworkInfo, error)
-
-	CreateEndpoint(networkId string, epInfo *EndpointInfo) error
-	DeleteEndpoint(networkId string, endpointId string) error
-	GetEndpointInfo(networkId string, endpointId string) (*EndpointInfo, error)
-	GetAllEndpoints(networkId string) (map[string]*EndpointInfo, error)
-	GetEndpointInfoBasedOnPODDetails(networkId string, podName string, podNameSpace string, doExactMatchForPodName bool) (*EndpointInfo, error)
-	AttachEndpoint(networkId string, endpointId string, sandboxKey string) (*endpoint, error)
-	DetachEndpoint(networkId string, endpointId string) error
-	UpdateEndpoint(networkId string, existingEpInfo *EndpointInfo, targetEpInfo *EndpointInfo) error
-	GetNumberOfEndpoints(ifName string, networkId string) int
+	DeleteNetwork(networkID string) error
+	GetNetworkInfo(networkID string) (NetworkInfo, error)
+
+	CreateEndpoint(client apipaClient, networkID string, epInfo *EndpointInfo) error
+	DeleteEndpoint(cli apipaClient, networkID string, endpointID string) error
```

Comment:
Do clients change on different invocations? Or are they tied to the lifetime of the manager? If the latter, this should probably just be an internal field of the manager.

Reply:
agree but refactoring the manager is wayyyy out of scope here imo and there is ongoing work for that happening elsewhere